addons/website_forum/tests/test_performance.py

```diff
@@ -0,0 +1,69 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command
+
+from odoo.addons.mail.tests.common import mail_new_test_user
+from odoo.addons.website.tests.test_performance import UtilPerf
+
+
+class TestForumPerformance(UtilPerf):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user_employee = mail_new_test_user(
+            cls.env,
+            login="employee",
+            groups="base.group_user",
+        )
+        cls.forum = cls.env['forum.forum'].create({'name': 'TestForum'})
+        cls.post = cls.env['forum.post'].create({
+            'name': 'TestQuestion',
+            'content': "Who's afraid of Virginia Wolf?",
+            'forum_id': cls.forum.id,
+            'tag_ids': [Command.create({'name': 'Tag2', 'forum_id': cls.forum.id})]
+        })
+
+    def test_perf_sql_forum_standard_data(self):
+        number_of_queries = self._get_url_hot_query(self.forum._compute_website_url())
+        self.assertEqual(number_of_queries, 24)
+        number_of_queries = self._get_url_hot_query(self.forum._compute_website_url(), cache=False)
+        self.assertLessEqual(number_of_queries, 28)
+        number_of_queries = self._get_url_hot_query(self.post.website_url)
+        self.assertEqual(number_of_queries, 20)
+        number_of_queries = self._get_url_hot_query(self.post.website_url, cache=False)
+        self.assertLessEqual(number_of_queries, 24)
+
+    def test_perf_sql_forum_scaling_answers(self):
+        self.env['forum.tag'].create([{
+            'name': f'Forum Post Test Tag {i}', 'forum_id': self.forum.id
+        } for i in range(20)])
+        answers = self.env['forum.post'].create([{
+            'name': f"TestAnswer {i}",
+            'content': "You",
+            'is_correct': i == 0,  # Ensure to have one accepted answer
+            'forum_id': self.forum.id,
+            'parent_id': self.post.id,
+        } for i in range(50)])
```

Comment:
:socks: some linting could be improved without impacting too much readability (and you can order dict keys alphabetically, allows to have a default ordering instead of random one :) )
```suggestion
        answers = self.env['forum.post'].create([
            {
                'content': f"AnswerContent {i}",
                'forum_id': self.forum.id,
                'is_correct': i == 0,  # Ensure to have one accepted answer
                'name': f"TestAnswer {i}",
                'parent_id': self.post.id,
            } for i in range(50)
        ])
```